### PR TITLE
Story-2: added files back

### PR DESCRIPTION
**Pull request by aatamaniv**

### CODE REVIEW

File: src/main/scala/com/atamaniv/Retry.scala [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package com.atamaniv
+
+import scala.concurrent.duration.FiniteDuration
+
+trait Retry {
+  def retry[A](block: () => A, acceptResult: A => Boolean, retries: List[FiniteDuration]): A
+}
```
Comment (azhur) [inline src/main/scala/com/atamaniv/Retry.scala]: why you need a trait and the implementation? why not just simple object with retry function?

File: src/main/scala/com/atamaniv/RetryImpl.scala [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+    if (acceptResult(result))
+      result
+    else retries match {
+      case head :: Nil =>
+        sleepFor(head.toMillis)
+        retry(block, acceptResult, Nil)
+      case head :: tail =>
+        sleepFor(head.toMillis)
+        retry(block, acceptResult, tail)
```
Comment (azhur) [inline src/main/scala/com/atamaniv/RetryImpl.scala]: what is the difference between 2 cases above? it looks like the first case is redundant

File: src/main/scala/com/atamaniv/RetryImpl.scala [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+      case Nil => throw new Throwable("Cannot retry any more")
```
Comment (azhur) [inline src/main/scala/com/atamaniv/RetryImpl.scala]: We talked about moving to pure functions, exceptions are side effects. Also I've explained that on the previous session: if all retries are exceeded then last block() result must be returned, no exceptions.

File: src/test/scala/com/atamaniv/RetryTest.scala [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  "A RetryImpl" should "throw an error then cannot retry execution any more" in {
+    val retryImpl = new RetryImpl
+    val intGenerator = () => 4
+    val isAcceptable: Int => Boolean = result => result > 5
+
+    assertThrows[Throwable] {
+      retryImpl.retry(intGenerator, isAcceptable, Nil)
+    }
+  }
+}
```
Comment (azhur) [inline src/test/scala/com/atamaniv/RetryTest.scala]: there are no tests that assert the number of retries the retry function does; there are no tests regarding the backoffs the retry function should wait for.

File: src/main/scala/com/atamaniv/Retry.scala (hunk 1 above)
Comment (aatamaniv) [reply src/main/scala/com/atamaniv/Retry.scala]: Thanks for review. I implemented your suggestion.

**Pull request merged**